lib/simple-dom/document/node.js

```diff
@@ -163,4 +162,21 @@ ChildNodes.prototype.item = function(index) {
   return child;
 };
 
+if (Object.defineProperty) {
```

Comment:
an extension.  There would be work to populate the cache overtime someone operates on the child. This is another reason `Proxy` is needed cc @moschel 

Proxy would allow you to trap the numeric based properties and fill them in.